subsys/net/lib/mqtt/mqtt.c

```diff
@@ -843,17 +786,7 @@ int mqtt_init(struct mqtt_ctx *ctx, enum mqtt_app app_type)
 	(void)net_context_recv(ctx->net_ctx, mqtt_recv, K_NO_WAIT, ctx);
 
 	ctx->app_type = app_type;
-
-	switch (ctx->app_type) {
-	case MQTT_APP_PUBLISHER:
-		ctx->rcv = mqtt_publisher_parser;
-		break;
-	case MQTT_APP_SUBSCRIBER:
-		ctx->rcv = mqtt_subscriber_parser;
-		break;
-	default:
-		return -EINVAL;
-	}
+	ctx->rcv = mqtt_parser;
```

Comment:
Yep, you're right.  I should probably put in a separate patch to fix that or add that fix first and then lay this patch on top.  If you want I can stick you in as author since it would be a code shuffle only patch (easy).